var/spack/repos/builtin/packages/orca/package.py

```diff
@@ -24,6 +24,11 @@ class Orca(Package):
 
     license("LGPL-2.1-or-later")
 
+    version(
+        "5.0.4",
+        sha256="c4ea5aea60da7bcb18a6b7042609206fbeb2a765c6fa958c5689d450b588b036",
+        url="file://{0}/orca_5_0_4_linux_x86-64_shared_openmpi411.tar.xz".format(os.getcwd()),
```

Comment:
Can this logic be added to `url_for_version` instead?

Reply:
The url for `4.0.1.2` should be left unchanged, but I'm in a bit of a roadblock.

The `4.X` versions, have their url pointing the static zst releases, zst is not much of a problem because the forum also offers xz releases for the same versions, the problem is that `4.X` are static releases and `5.X` are dynamic, (I blindly followed what was done for `5.0.3` and chose the dynamic releases because `4.X` were broken anyways), but how should we deal with this inconsistency?

I'm new to packaging and I've never seen what was done in this situation but should we
- split static and dynamic releases for all available versions? (how?)
  For this we would need to check multiple checksums for `5.0.3` and `5.0.4` because their static releases are packaged in 3 files.
- stick to either static or dynamic releases? If we choose static we would need to solve the same problem mentioned before.

What I would do for now is to fix a `url_for_version` that works for the current (dynamic) `5.0.3` and `5.0.4` (which are the only ones that work out of the box) and maybe open a discussion about what to do with `4.X` and dynamic vs static releases.

Reply:
@alvaro-sch this package was originally created with the forum versions in mind, I believe. For a while there we also had a patched version included, but that was moved into its own package.

The archives that were hashed for the older releases are the zst compressed ones, which has caused confusion in the past. I am not opposed to switching to the xz compressed ones; zst may have been the only offering when this was originally written, I do not remember. That would also remove the zst dependency.

I am pretty sure the 4.x versions point at the same style of dynamic linked releases as the 5.x releases. They may have changed the name of the archives at some point, I've not checked in a while.

Anyway, for a version bump this PR as is is fine, if you want to rework it to use the xz archives for all versions and remove the (seemingly unavailable) 4.0.1.2 version feel free.

Reply:
Great! I'll leave the urls for `4.X` untouched for now. Fixing them should be easy since (as of now) they follow the same naming standard as `5.X` zst can also be replaced with xz, but that should be a different PR.

Reply:
I'd rather you just go with your original commit to this. And save significant changes for a separate pr.